### ISSUE TITLE
moment().format('yyyy') produces unexpected results.

### ISSUE THREAD

**Issue by bzd3y** [moment/moment#5649]
**The bug:**
When calling `.format('yyyy')` (or any format string that contains 'yyyy' as far as I can tell) on a moment object, the moment object's time is set to midnight afterward. This happens regardless of the client's time zone as far as I can tell.

Also, if the client's time zone changes after (I assume) moment has been initialized, the 4-digit year returned from `.format('yyyy')` may also be incorrect.

Now, I realize the documents say to use 'YYYY' to return the 4-digit year, but it still seems like the moment functions operate on the object as if they are immutable and would not change the current object. But maybe it is expected/acceptable in this case? If that's the case then this bug can be ignored/deleted.

**To reproduce:**
Go to https://momentjs.com/
Run this in the Developer Tools console:
```
var test = moment(new Date());
console.log(test.format());
console.log(test.format('yyyy'));
console.log(test.format());
```
The above code prints out:
2020-07-10T21:41:33-04:00
2020
2020-07-10T**00:00:00**-04:00

After switching to UTC time zone:
2020-07-11T01:57:32+00:00
**2021**
2020-07-11T00:00:00+00:00

The most reproducible sequence I came up with is that if I set my time zone to +05:45 (Kathmandu) and run the above, I get '2020'. If I then switch to UTC I get '2021'. If I then switch to EST/EDT I get '2021' again.

**Expected behavior**
The time for `test` should not be altered after the call to `format('yyyy')`. The 4-digit year should always be correct for the client's time zone.

**Browsers:**
I am seeing this in:

- Google Chrome 83.0.4103.116
- Microsoft Edge (Chromium) 83.0.478.61
- Firefox 78.0.2

**Moment-specific environment**

* The time zones do not appear to matter. I observed this mostly in EST/EDT and UTC.
* The code above was run on the date(s) above: 2020-07-10T21:41:33-04:00
* Other libraries include whatever momentjs.com is using.

Please run the following code in your environment and include the output:
```
console.log((new Date()).toString())
console.log((new Date()).toLocaleString())
console.log((new Date()).getTimezoneOffset())
console.log(navigator.userAgent)
console.log(moment.version)
```
Fri Jul 10 2020 22:05:45 GMT-0400 (Eastern Daylight Time)
7/10/2020, 10:05:45 PM
240
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36
2.27.0

**Comment by MinusFour** [comment]
This is a duplicate of #5571. There is a PR to fix it that seems ok to me.

**Comment by bzd3y** [comment]
Oh, okay. Thanks. I tried to search for an existing issue, but didn't see that one.

**Comment by marwahaha** [comment]
This is fixed in 2.28.0
